src/dotnet/GeneratorAgent/src/Azure.Tools.GeneratorAgent/Services/GeneratorAgentService.cs

```diff
@@ -0,0 +1,90 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.AI.Agents.Persistent;
+using Azure.Identity;
+using Azure.Tools.GeneratorAgent.Interfaces;
+
+namespace Azure.Tools.GeneratorAgent.Services
+{
+    public class GeneratorAgentService : IAsyncDisposable
```

Comment:
```suggestion
    public class GeneratorAgent : IAsyncDisposable
```
I'd avoid "Service" here, as this doesn't provide a set of operations that someone can call in a specific area, it just gets created and then does its own thing.

Reply:
Renamed!!